iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java

```diff
@@ -1463,10 +1482,26 @@ private void resetWithNewestRebootTime(int connectorRebootTimes) {
       LOGGER.info(
           "PipeConsensus-PipeName-{}: receiver detected an newer rebootTimes, which indicates the leader has rebooted. receiver will reset all its data.",
           consensusPipeName);
+      // since pipe task will resend all data that hasn't synchronized, it's safe to clear all
+      // events in buffer.
       this.reqExecutionOrderBuffer.clear();
       this.onSyncedCommitIndex = 0;
       // sync the follower's connectorRebootTimes with connector's actual rebootTimes
       this.connectorRebootTimes = connectorRebootTimes;
+      // Note: dataNode reboot will reset pipeTaskRestartTimes
+      this.pipeTaskRestartTimes = 0;
+      this.tsFileWriterPool.handleExit(consensusPipeName);
+    }
+
+    private void resetWithNewestRestartTime(int pipeTaskRestartTimes) {
+      LOGGER.info(
+          "PipeConsensus-PipeName-{}: receiver detected an newer pipeTaskRestartTimes, which indicates the pipe task has restarted. receiver will reset all its data.",
+          consensusPipeName);
+      // since pipe task will resend all data that hasn't synchronized, it's safe to clear all
+      // events in buffer.
+      this.reqExecutionOrderBuffer.clear();
+      this.onSyncedCommitIndex = 0;
+      this.pipeTaskRestartTimes = pipeTaskRestartTimes;
       this.tsFileWriterPool.handleExit(consensusPipeName);
```

Comment:
How about writing it like this?

`    private void resetRebootTime(int rebootTime) {
      this.connectorRebootTimes = rebootTime;
      clear();
    }

    private void resetPipeRestartTime(int restartTime) {
      this.pipeTaskRestartTime = restartTime;
      clear();
    }

    private void clear() {
      this.reqExecutionOrderBuffer.clear();
      this.tsFileWriterPool.handleExit(consensusPipeName);
      this.onSyncedCommitIndex = 0;
    }`

Reply:
fixed